src/core/Table/Table.tsx

```diff
@@ -259,6 +274,12 @@ export const Table = <
       case TableActions.setFilter:
         onFilterHandler(newState, action, previousState, instance);
         break;
+      case TableActions.toggleRowSelected:
+      case TableActions.toggleAllRowsSelected:
+      case TableActions.toggleAllPageRowsSelected: {
+        onSelectHandler(newState, instance);
```

Comment:
Have you thought about how this would work for controlled selections?

Reply:
This is just a callback when user selects something. 
I have started investigating `useControledState` prop and it still works - it allows to control what rows should be selected. Also, initial state also works.

Reply:
![table-controlled-state](https://user-images.githubusercontent.com/36186912/122410622-c3a9fa80-cf8c-11eb-81c4-5a9bb553dd2d.gif)

Reply:
Ok great! Just wanted to make sure it doesn't need additional refactoring when we add those in the future 😄